src/services/services.ts

```diff
@@ -7866,6 +7886,30 @@ namespace ts {
             return [];
         }
 
+        function getCodeFixesAtPosition(fileName: string, start: number, end: number, errorCodes: string[]): CodeAction[] {
+            synchronizeHostData();
+            const sourceFile = getValidSourceFile(fileName);
+            const checker = program.getTypeChecker();
+            let allFixes: CodeAction[] = [];
+
+            forEach(errorCodes, error => {
+                const context = {
+                    errorCode: error,
+                    sourceFile: sourceFile,
+                    span: { start, length: end - start },

```

Comment:
define the span outside the loop and reuse it. no reason to create a new one every time. also use `createTextSpanFromBounds` instead.